src/App.js

```diff
@@ -0,0 +1,227 @@
+        }).then(res => res.json())
+            .then(console.log)
+            .then(() => this.fetchDataHandler());
+    };
+
+
+    putTodoStatusHandler = (todo_id, current_content, done_stats) => {
+        fetch('http://localhost:3000/todos/' + todo_id, {
+            method: 'PUT',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify({
+                'content': current_content,
+                'done': !done_stats
+            })
+        }).then(res => res.json())
+            .then(console.log)
+            .then(() => this.fetchDataHandler());
+    };
+
+    deleteTodoHandler = (todo_id) => {
+        fetch('http://localhost:3000/todos/' + todo_id, {
+            method: 'DELETE'
+        }).then(() =>this.fetchDataHandler())
+            .then(console.log)
+
+    };
+
+
+//Todo: fetchを調べてフィルター機能を作り、done=trueのものだけ取り出す。
+
+    deleteCompletedTasksHandler = () => {
+        const copiedTodos = [...this.state.todos];
+        const updateTodos = copiedTodos.filter(todo => {
+            return todo.done === false;
+        });
+        this.setState({todos: updateTodos});
+    };
+
+    // below methods are used with state.
+
+    // addTodoHandler = (content_changed) => {
+    //     console.log("inside of add todo");
+    //
+    //     this.setState({
+    //         todos: [...this.state.todos,
+    //             {
+    //                 id: this.state.nextId + 1,
+    //                 content: content_changed,
+    //                 done: false
+    //             }
+    //         ],
+    //         nextId: this.state.nextId + 1
+    //     });
+    // };
+
+
+    // deleteTodoHandler = (id) => {
+    //     const copiedTodos = [...this.state.todos];
+    //     const updateTodos = copiedTodos.filter(todo => {
+    //         return todo.id !== id
+    //     });
+    //     this.setState({todos: updateTodos});
+    // };
+
+
+    // toggleStatusHandler = (id) => {
+    //
+    //     //find todoItem which I want to change
+    //     const todoIndex = this.state.todos.findIndex(todo => {
+    //         return todo.id === id;
+    //     });
+    //
+    //     const updatedTodo = {
+    //         ...this.state.todos[todoIndex]
+    //     };
+    //
+    //     //change done status
+    //     const currentDoneStatus = updatedTodo.done; //false
+    //
+    //     updatedTodo.done = !currentDoneStatus; //false -> true
+    //
+    //     //save the change above safely
+    //     const todos = [...this.state.todos];
+    //     todos[todoIndex] = updatedTodo;
+    //     this.setState({todos: todos});
+    // };
+
+
+    render() {
+        //
+        const completedTodos = this.state.todos.filter(todo => {
+            return todo.done === true;
+        });
+
+        const notCompletedTodos = this.state.todos.filter(todo => {
+            return todo.done === false;
+        });
+
+        const completedTodosNum = completedTodos.length;
```

Comment:
Do you need completedTodosNum ?
I guess doing `completedTodos.length`  in each component is easy to understand, and not so much extra code.

Reply:
Apparently not, this code was potato... Thank you!

Reply:
potato!?

Reply:
It is just a slang tho haha
https://www.urbandictionary.com/define.php?term=potato

>A person who is brainless and unaware of anything.

>An object of poor quality